files/ruby_plugin_helper.rb

```diff
@@ -0,0 +1,48 @@
+TARGET_OPTS = %i[uri name config].freeze
+
+def config_lookup(opts)
```

Comment:
global functions mean this will be harder to unit test.

As a general rule global functions are evil and should only used in a script that is never tested or loaded into another file.

Reply:
I don't think we can have plugin tasks inherit from both the ruby task helper and the plugin task helper (which we would need to do for all of our current plugins). I don't think combining them into one helper makes sense. I could make this a module and `include` that, but I'm not sure how that's meaningfully different from what's here.

Reply:
that is what `include` is for.